### ISSUE TITLE
Duplicate `Put` in point slice pool from `mergeSeries`

### ISSUE THREAD

**Issue by shanson7** [grafana/metrictank#1875]
**Describe the bug**

In certain scenarios during a render request a slice can be returned to the pool 2 times. We saw corrupted responses because of this and so deployed a change that (within a single render) detects this case, does not return the slice to the pool twice and logs additional information. With this info I think I've found where it happens. In the (likely rare) case where there are identical time-series with 2 different intervals, [`mergeSeries`](https://github.com/grafana/metrictank/blob/001015d92537c73fedff18e3bb473f947bd1c80e/api/dataprocessor.go#L658) will merge them together to form a single normalized value. 

It then puts all the other series back into the pool and only keeps the first one. However, [`normalize`](https://github.com/grafana/metrictank/blob/001015d92537c73fedff18e3bb473f947bd1c80e/expr/normalize.go#L78) already puts the new normalized value into the datamap. So, in the scenario where there is series `foo.bar` with intervals 60 (let's call this `foo60`) and 30 (`foo30`) (in this order), the process flow goes something like:

1. `mergeSeries` calls `normalize(datamap, [foo60, foo30])`
1. `normalize` keeps `foo60` (nothing to do)
1. `normalize` copies `foo30`, consolidates it to 60 seconds and adds it to the datamap (let's call this `foo60_2`)
   - Note: At this point we dropped the slice from `foo30` to be GC'd. It did not go into the pool
1. `mergeSeries` then iterates over the points in `foo60` checking if it can fill `NaN` values from `foo60_2`
1. Finally, `mergeSeries` adds `foo60_2.Datapoints` into the pool, even though it is already in the datamap!

**Helpful Information**
Metrictank Version: ` v1.0-52` (custom build off of commit e0850173eac8153aa7dbb08d9de7b2ac12b529c8)
Golang Version:  `go1.12.7`

**Comment by shanson7** [comment]
For completeness, the reverse case (`[foo30, foo60]`) is also an issue:

1. `mergeSeries` calls `normalize(datamap, [foo30, foo60])`
1. `normalize` keeps `foo60` (nothing to do)
1. `normalize` copies `foo30`, consolidates it to 60 seconds and adds it to the datamap (let's call this `foo60_2`)
   - Note: At this point we dropped the slice from `foo30` to be GC'd. It did not go into the pool
1. `mergeSeries` then iterates over the points in `foo60_2` checking if it can fill `NaN` values from `foo60_2`
1. `mergeSeries` adds `foo60.Datapoints` into the pool (this is ok)
1. [`executePlan`}(https://github.com/grafana/metrictank/blob/001015d92537c73fedff18e3bb473f947bd1c80e/api/graphite.go#L830) adds `foo60_2` to the datamap for a second time.

This version is the one that the code I added to our rollout is actually detecting. The one in the original description won't be detected and the pool will be corrupted until next clean or until detected.

**Comment by Dieterbe** [comment]
Very nice catch!

The expr.Normalize functions were originally meant to be used within the expr package, inside of a plan.Run() call.
that means they should follow this regimen:
* any unused series should be left alone (may be referenced or read from later, e.g. different function processing chain)
* any newly created series requires an entry in datamap, to make sure it'll be reclaimed later

When we added normalization into mergeSeries - in #1674 - we reused the same expr.Normalize functions but from a different call site, where the expectations are different (e.g. where we will still add all our series to the datamap, and thus potentially add them twice)

Let me dig a bit more into this and figure out a clean way to resolve this
